### ISSUE TITLE
linter: false positive for `no-invalid-void-type` rule

### ISSUE THREAD

**Issue by rtarpine-vanta** [oxc-project/oxc#20777]
### What version of Oxlint are you using?

1.57.0

### What command did you run?

oxlint $file

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "no-invalid-void-type": "error"
  }
}
```

### What happened?

The following code triggers a `typescript-eslint(no-invalid-void-type): Use `void` only as a return type or generic type argument` error:
```ts
interface Producer<T> {
  get: () => T;
}

export class Test implements Producer<void> {
  get = () => null;
}
```
But the original `no-invalid-void-type` rule doesn't complain since `void` is used as a generic type argument.
